src/V3Const.cpp

```diff
@@ -76,6 +76,418 @@ class ConstVarFindVisitor final : public AstNVisitor {
     bool found() const { return m_found; }
 };
 
+// This visitor can be used expanded Ast by V3Expand, such Ast satisfies:
+// - Constants are 64 bit at most (because words are accessed via AstWordSel)
+// - Variables are scoped.
+class ConstBitOpTreeVisitor final : public AstNVisitor {
+    // TYPES
+
+    struct LeafInfo {  // Leaf node (either AstConst or AstVarRef)
+        bool m_polarity = true;
+        int m_lsb = 0;
+        int m_wordIdx = -1;  // -1 means AstWordSel is not used.
+        AstVarRef* m_refp = nullptr;
+        AstConst* m_constp = nullptr;
+    };
+    // Collect information for each Variable to transform as below
+    class VarInfo final {
+        // MEMBERS
+        int m_constResult = -1;  // -1: result is not constant, 0 or 1: result of this tree
+        ConstBitOpTreeVisitor* m_parent;
```

Comment:
```suggestion
        ConstBitOpTreeVisitor* m_parentp;
```

and //comment please

Reply:
Renamed in acef8e00934ba1dcc1510102282071d94118de35